src/app/loadout/loadout-menu/LoadoutPopup.tsx

```diff
@@ -384,3 +334,68 @@ function filterLoadoutToEquipped(loadout: Loadout) {
     items: loadout.items.filter((i) => i.equip),
   };
 }
+
+function RandomLoadoutButton({ store, query }: { store: DimStore; query: string }) {
+  const defs = useDefinitions()!;
+  const dispatch = useThunkDispatch();
+  const allItems = useSelector(allItemsSelector);
+  const searchFilter = useSelector(searchFilterSelector);
+  const unlockedPlugs = useSelector(unlockedPlugSetItemsSelector(store.id));
+
+  const createRandomLoadout = useCallback(() => {
```

Comment:
This could be a redux thunk-action that fishes all that selector stuff out when it's invoked rather than having to take them via `useSelector`.

Reply:
Fixed, I think?